Description: Fix memory leak by freeing strdup-ed string
Forwarded: yes
Author: Rowan Thorpe <rowan@rowanthorpe.com>
Last-Update: 2014-03-03
--- a/src/sflowtool.c
+++ b/src/sflowtool.c
@@ -4428,9 +4428,11 @@
     else sfConfig.readPcapFile = fopen(sfConfig.readPcapFileName, "rb");
     if(sfConfig.readPcapFile == NULL) {
       fprintf(ERROUT, "cannot open %s : %s\n", sfConfig.readPcapFileName, strerror(errno));
+      free(sfConfig.readPcapFileName);
       exit(-1);
     }
     readPcapHeader();
+    free(sfConfig.readPcapFileName);
   }
   else {
     /* open the input socket -- for now it's either a v4 or v6 socket, but in future
